src/storage/raw/raw_mvcc.rs

```diff
@@ -0,0 +1,244 @@
+        self.is_valid = Some(true);
+    }
+
+    fn clear_cur_kv(&mut self) {
+        self.cur_key = None;
+        self.cur_value = None;
+        self.is_valid = None;
+    }
+
+    fn move_to_prev_max_ts(&mut self) -> Result<bool> {
+        if self.inner.valid()? {
+            self.update_cur_kv();
+            self.inner.prev()?;
+        } else {
+            self.clear_cur_kv();
+            return Ok(false);
+        }
+        while self.inner.valid()? {
+            // cur_key should not be None here.
+            if is_user_key_eq(self.cur_key.as_ref().unwrap(), self.inner.key()) {
+                self.update_cur_kv();
+                self.inner.prev()?;
+            } else {
+                break;
+            }
+        }
+        Ok(true)
+    }
+}
+
+// RawMvccIterator always return the latest ts of user key.
+// ts is desc encoded after user key, so it's placed the first one for the same user key.
+// Only one-way direction scan is supported. Like `seek` then `next` or `seek_for_prev` then `prev`
+impl<I: Iterator> Iterator for RawMvccIterator<I> {
+    fn next(&mut self) -> Result<bool> {
+        if !self.is_forward {
+            return Err(Error::from(ErrorInner::Other(Box::from(
+                "invalid raw mvcc operation",
+            ))));
+        }
+        let cur_key = self.inner.key().to_owned();
+        let mut res = self.inner.next()?;
+        while res && self.inner.valid()? && is_user_key_eq(&cur_key, self.inner.key()) {
+            res = self.inner.next()?;
+        }
+        self.clear_cur_kv();
+        Ok(res)
+    }
+
+    fn prev(&mut self) -> Result<bool> {
+        if self.is_forward {
+            return Err(Error::from(ErrorInner::Other(Box::from(
+                "invalid raw mvcc operation",
+            ))));
+        }
+        self.move_to_prev_max_ts()
+    }
+
+    fn seek(&mut self, key: &Key) -> Result<bool> {
+        self.is_forward = true;
+        self.clear_cur_kv();
+        self.inner.seek(key)
+    }
+
+    fn seek_for_prev(&mut self, key: &Key) -> Result<bool> {
+        self.is_forward = false;
+        if self.inner.seek_for_prev(key)? {
+            self.move_to_prev_max_ts()
+        } else {
+            Ok(false)
+        }
+    }
+
+    fn seek_to_first(&mut self) -> Result<bool> {
+        self.is_forward = true;
+        self.clear_cur_kv();
+        self.inner.seek_to_first()
+    }
+
+    fn seek_to_last(&mut self) -> Result<bool> {
+        self.is_forward = false;
+        if self.inner.seek_to_last()? {
+            self.move_to_prev_max_ts()
+        } else {
+            Ok(false)
+        }
+    }
+
+    fn valid(&self) -> Result<bool> {
+        self.is_valid.map_or_else(|| self.inner.valid(), Ok)
+    }
+
+    fn validate_key(&self, key: &Key) -> Result<()> {
+        self.inner.validate_key(key)
+    }
+
+    fn key(&self) -> &[u8] {
+        // need map_or_else to lazy evaluate the default func, as it will abort when invalid.
+        self.cur_key
```

Comment:
cur_kv seems unnecessary, why not just use inner.key() and inner.value() directly?